docs/userguide/glossary.rst

```diff
@@ -0,0 +1,218 @@
+
+.. _configuration:
+
+**Configuration:**
+---------------------
+
+Configuration is like setting up the rules for how Parsl should work. It's like adjusting the settings on your phone – you can choose how you want things to look and behave. In Parsl, you can configure things like how many computers to use, where to store data, and how to handle errors.
+
+.. _datafuture:
+
+**DataFuture:**
+------------------
+
+A DataFuture is like a placeholder for a file that an app is creating. It's like a receipt for a package you're expecting – you get the receipt right away, but you have to wait for the package to arrive. Similarly, you get a DataFuture immediately when an app starts creating a file, but you have to wait for the file to be finished before you can use it.
+
+.. _dfk:
+
+**DataFlowKernel (DFK):**
+------------------------------
+
+The DataFlowKernel is like the brain of Parsl. It's the part that controls how your apps run and how they share information. It's like the conductor of an orchestra, making sure that all the musicians play together in harmony.
+
+.. _elasticity:
+
+**Elasticity:**
+-----------------
+
+Elasticity means being able to stretch or shrink. In Parsl, it means that you can easily add or remove blocks of computers as needed. This is useful if your workload changes – you can use more computers when you have a lot of work to do and fewer computers when you don't.
+
+.. _executionprovider:
+
+**Execution Provider:**
+--------------------------
+
+An execution provider is like a bridge between Parsl and the computers you want to use. It's the part that knows how to talk to different types of computers, like your laptop, a cluster, or a cloud service.
+
+.. _executor:
+
+**Executor:**
+----------------
+
+An executor is like a manager for your apps. It's the part that decides which app should run on which computer and when. It's like a traffic controller, directing the flow of apps to make sure they all get where they need to go.
+
+.. _future:
+
+**Future:**
+-------------
+
+A future is a placeholder for the result of a task that hasn't finished yet. Both AppFuture and DataFuture are types of Futures. You can use the `.result()` method to get the actual result when it's ready.
+
+.. _job:
+
+**Job:**
+---------
+
+A job in Parsl refers to a unit of work that is submitted to an execution environment (like a cluster or cloud) for processing. It's like a task that needs to be done, such as running a script or processing data, and it can consist of one or more tasks that are executed on a compute resource.
+
+.. _launcher:
+
+**Launcher:**
+----------------
+
+A launcher is like a starter for your apps. It's the part that actually starts the apps running on the computers. It's like a coach telling the athletes when to start running.
```

Comment:
A point of clarity: the Launcher doesn't actually start the apps. It puts the workers onto each computer. So, more like the bus driver who brings the players to a stadium than the coach who tells them what to do.